checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt

```diff
@@ -0,0 +1,439 @@
+            """
+            package clean
+
+            import java.io.File
+            import assertk.assertThat
+            import assertk.key
+            import assertk.isNotNull
+
+            class TestingTesting {
+                fun testingTest() {
+                    val map: Map<String, String?> = mapOf("9A3E6FAC-0639-4F52-8E88-D9F7512540A4" to "John")
+
+                    assertThat(map.getOrDefault("9A3E6FAC-0639-4F52-8E88-D9F7512540A4", null)).isNotNull()
+                }
+            }
+            """.trimIndent()
+
+        lint().files(kotlin(code), kotlin(assertkStub)).run().expect(
+            """src/clean/TestingTesting.kt:12: Warning: Use Assert.key for map entries [MapValueAssertion]
+        assertThat(map.getOrDefault("9A3E6FAC-0639-4F52-8E88-D9F7512540A4", null)).isNotNull()
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+0 errors, 1 warnings""",
+        )
+    }
+
+    @Test
+    fun `getOrElse function read from map in assertion subject creation detected`() {
+        val code =
+            """
+            package clean
+
+            import java.io.File
+            import assertk.assertThat
+            import assertk.key
+            import assertk.isNotNull
+
+            class TestingTesting {
+                fun testingTest() {
+                    val map: Map<String, String?> = mapOf("9A3E6FAC-0639-4F52-8E88-D9F7512540A4" to "John")
+
+                    assertThat(map.getOrElse("9A3E6FAC-0639-4F52-8E88-D9F7512540A4") { error("Nope") }).isNotNull()
+                }
+            }
+            """.trimIndent()
+
+        lint().files(kotlin(code), kotlin(assertkStub)).run().expect(
+            """src/clean/TestingTesting.kt:12: Warning: Use Assert.key for map entries [MapValueAssertion]
+        assertThat(map.getOrElse("9A3E6FAC-0639-4F52-8E88-D9F7512540A4") { error("Nope") }).isNotNull()
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+0 errors, 1 warnings""",
+        )
+    }
+
+    @Test
+    fun `map keys read for checking key not present`() {
+        val code =
+            """
+            package clean
+
+            import java.io.File
+            import assertk.assertThat
+            import assertk.assertions.doesNotContain
+
+            class TestingTesting {
+                fun testingTest() {
+                    val map: Map<String, String?> = mapOf("9A3E6FAC-0639-4F52-8E88-D9F7512540A4" to "John")
+
+                    assertThat(map.keys).doesNotContain("")
+                }
+            }
+            """.trimIndent()
+
+        lint().files(kotlin(code), kotlin(assertkStub), kotlin(assertkCollectionStub)).run().expect(
+            """src/clean/TestingTesting.kt:11: Warning: Use Assert.doesNotContainKey to assert absence [KeySetAbsentAssertion]
+        assertThat(map.keys).doesNotContain("")
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+0 errors, 1 warnings""",
+        )
+    }
+
```

Comment:
Do you care about usages like `map.entries.map { it.key }`?

Reply:
Maybe. Have you seen that call before in the wild?

It strikes me as odd to iterate the entries to read a set that the map is probably already maintaining internally, but if people do that for whatever reason…

Reply:
Ha, honestly I'm not sure; I was mostly trying to find edge cases related to keys. 

I think I've written it before, but probably incorrectly 🙂

Reply:
I'm going to merge this, but I created #11 to track interest in something like that over time. I explained a little of my rationale there.